superset/common/query_context_processor.py

```diff
@@ -322,6 +322,16 @@ def processing_time_offsets(  # pylint: disable=too-many-locals
                 #  multi-dimensional charts
                 granularity = query_object.granularity
                 index = granularity if granularity in df.columns else DTTM_ALIAS
+                if not pd.api.types.is_datetime64_any_dtype(
```

Comment:
Based on recent errors I'm guessing this is too restrictive.

Reply:
Guessing this should be `is_datetime64_any_dtype(...) or is_datetime64tz_any_dtype(...)`

Reply:
<img width="563" alt="Screen Shot 2022-07-05 at 12 32 20 PM" src="https://user-images.githubusercontent.com/487433/177402252-71ee5013-f129-462e-8d2f-258473446c2a.png">

Reply:
I will open a new PR and depend on the Generic Type of column instead of Pandas type.

Reply:
`is_datetime64_any_dtype` verify `is_datetime64tz_any_dtype` as well, https://github.com/pandas-dev/pandas/blob/e8093ba372f9adfe79439d90fe74b0b5b6dea9d6/pandas/core/dtypes/common.py#L922